tuf/repository_tool.py

```diff
@@ -3104,45 +3104,24 @@ def load_repository(repository_directory, repository_name='default',
   # extracted fileinfo is stored in the 'meta' field of the snapshot metadata
   # object.
   targets_objects = {}
-  loaded_metadata = []
   targets_objects['targets'] = repository.targets
+  # A list of delegated-delegating role pairs
+  delegations = []
 
-  metadata_files = sorted(storage_backend.list_folder(metadata_directory),
-      reverse=True)
-  for metadata_role in metadata_files:
+  delegations_filenames = repo_lib.get_delegations_filenames(metadata_directory,
+        consistent_snapshot, storage_backend)
 
-    metadata_path = os.path.join(metadata_directory, metadata_role)
-    metadata_name = \
-      metadata_path[len(metadata_directory):].lstrip(os.path.sep)
+  # Top-level roles are already loaded, fetch targets and get its delegations.
+  # Collect a list of delegated-delegating role pairs, starting from the
+  # top-level targets: [('role1', 'targets'), ('role2', 'targets'), ... ]
+  roleinfo = tuf.roledb.get_roleinfo('targets', repository_name)
+  for role in roleinfo['delegations']['roles']:
+    delegations.append([role['name'], 'targets'])
```

Comment:
Minor nit: The comment suggests delegations are list of tuples, but the code makes it a list of lists. Please change either one. IMO tuples are fine.

Reply:
And tuples they are!